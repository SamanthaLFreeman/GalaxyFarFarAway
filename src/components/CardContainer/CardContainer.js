import React from 'react';
import Card from '../Card/Card'
import './CardContainer.scss';

const CardContainer = ({allData, toggleFavorite}) => {
  // console.log(allData.toggleFavorite)
  const cards = allData.map(data => {
    const {name, terrain, population, climate, residents, homeworld, species, model, vehicleClass, numOfPassengers} = data;
    return <Card 
    name={name}
    terrain={terrain}
    population={population}
    climate={climate}
    residents={residents}
    homeworld={homeworld}
    species={species}
    model={model}
    vehicleClass={vehicleClass}
    numOfPassengers={numOfPassengers}
    toggleFavorite={toggleFavorite}
    key={name}/>
  })

  return (
    <section className='card-container'>
      {cards}
    </section>
  )
}

export default CardContainer;