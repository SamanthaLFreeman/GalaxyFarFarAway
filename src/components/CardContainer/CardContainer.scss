.card-container {
    margin-top: 40px;
}

h1 {
    font-size: 3em;
}

.no-favs-message {
    color: aliceblue;
    font-weight: bold;
    text-align: center;
    padding: 40px;
    background-color: #2d445b;
}