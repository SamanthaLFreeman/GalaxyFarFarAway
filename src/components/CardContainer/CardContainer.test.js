import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const data = [
      {
        name: 'Brandy',
        homeworld: 'Earth',
        poupaltion: 'a lot',
        species: 'Human',
        language: 'English',
        isFavorite: true
      }
    ]
    const wrapper = shallow(<CardContainer allData={data} toggleFavorite={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });
});