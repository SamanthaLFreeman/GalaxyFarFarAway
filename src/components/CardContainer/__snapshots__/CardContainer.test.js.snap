// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="card-container"
>
  <Card
    homeworld="Earth"
    isFavorite={true}
    key="Brandy"
    language="English"
    name="Brandy"
    species="Human"
    toggleFavorite={[MockFunction]}
  />
</section>
`;
